raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -132,8 +133,13 @@ public long timeUntilCheckQuorumExpires(long currentTimeMs) {
      */
     public void updateCheckQuorumForFollowingVoter(int id, long currentTimeMs) {
         updateFetchedVoters(id);
-        // The majority number of the voters excluding the leader. Ex: 3 voters, the value will be 1
-        int majority = voterStates.size() / 2;
+        // The majority number of the voters. Ex: 2 for 3 voters, 3 for 4 voters... etc.
+        int majority = (int) Math.ceil((double) (voterStates.size() + 1) / 2);
```

Comment:
I don't trust conversions to `double` since they are not exact integers. I think you want:
```java
int majority = (voterState.size() / 2) + 1;
```

Reply:
That works, too. Updated.